code-challenges.test.js

```diff
@@ -14,38 +14,166 @@
 
 // b) Create the function that makes the test pass.
+const voteTally = (votes) => {
+  return votes.upVotes - votes.downVotes
+
+} //output:
+//   PASS  ./code-challenges.test.js
+//   PASS  ./jest.test.js
+ 
+//  Test Suites: 2 passed, 2 total
+//  Tests:       4 passed, 4 total
+//  Snapshots:   0 total
+//  Time:        0.138 s, estimated 1 s
+
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
-// a) Create a test with an expect statement using the variables provided.
+  // Pseudocode:
+    // Given 2 arrays 
+    // Create describe function named mergeArrays. In that function, we will define the expected output, the result, and the given arrays
+    // Use the spread operator to put all of the values from the arrays into a single array
+    // Use .concat() to add both arrays into a single array called "mergedArray"
+    // Use Array.from to to create a new copy of the array that we concatted
+    // Use "Set" to automatically remove the duplicate values in mergedArray
+    // Assign the new array to 'differentArray'
+    // Output will be the expected values
 
-const dataArray1 = ["array", "object", "number", "string", "Boolean"]
-const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
-// Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+describe("mergeArrays", () => {
+  it("merges and removes duplicates from two arrays", () => {
+    const dataArray1 = ["array", "object", "number", "string", "Boolean"]
+    const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
+    const result = mergeArrays(dataArray1, dataArray2);
+    const expected = ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+        expect(result).toEqual(expected)
+      })
+    }) //output:  FAIL  ./code-challenges.test.js
+    // ● mergeArrays › merges and removes duplicates from two arrays
+
+    //   ReferenceError: mergeArrays is not defined
+
+    //     111 |     const dataArray1 = ["array", "object", "number", "string", "Boolean"]
+    //     112 |     const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
+    //   > 113 |     const result = mergeArrays(dataArray1, dataArray2);
+    //         |                    ^
+    //     114 |     const expected = ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+    //     115 |     expect(result).toEqual(expected)
+    //     116 |   })
+
+    //     at Object.mergeArrays (code-challenges.test.js:113:20)
+
+    // b) Create the function that makes the test pass.
+    const mergeArrays = (...arrays) => {
+      const mergedArray = [].concat(...arrays)
+      const differentArray = Array.from(new Set(mergedArray))
+      return differentArray;
+    } //output:  
+    // PASS ./code-challenges.test.js
```

Comment:
Very nice! Great use of `Array.from`, `...`, and `new Set`. Also great job on using the spread operator in the parameter field to allow as many arguments as you want!

Reply:
I was hoping that I had it right. This one stumped me for an entire day lol